### ISSUE TITLE
First initialize the Stripe Terminal SDK before performing any action"

### ISSUE THREAD

**Issue by MoinJanjua** [stripe/stripe-terminal-react-native#749]
**Describe the bug**

I am trying to implement Stripe terminal sdk but i ma unable to initialize it I have below error
"First initialize the Stripe Terminal SDK before performing any action 
    at CartReaders"

**To Reproduce**
I Create the Root,App and Payment screen After running  Root and App screen and open the Payment screen to display readers than it display error  even I will try with connection_token directly pass to function
"First initialize the Stripe Terminal SDK before performing any action 
    at CartReaders     (http://localhost:8081/index.bundle//&platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=com.grunq.pos:197064:26)"

**Expected behavior**
I want it should initialize the SDK properly

**Screenshots**
![Screenshot 2024-07-04 at 3 06 21 PM](https://github.com/stripe/stripe-terminal-react-native/assets/52666065/3341e383-3a5a-4b21-8124-8984f73c2299)
![Screenshot 2024-07-04 at 3 10 25 PM](https://github.com/stripe/stripe-terminal-react-native/assets/52666065/6d64cbdd-437d-4b9b-acbd-de9319fbf8b7)

**Stripe Terminal React Native SDK version**

   "@stripe/stripe-terminal-react-native": "^0.0.1-beta.19",

**Smartphone (please complete the following information):**

- Device: [e.g. iPad7]
- OS: [e.g. iOS 17.0]

**Additional context**

**Comment by nabilfreeman** [comment]
Did you wrap the application with `StripeTerminalProvider` and hook up the server-side connection token API endpoint properly?

**Comment by MoinJanjua12** [comment]
Yes I did properly even i try with Static Token that i generated from the Postman

**Comment by nabilfreeman** [comment]
I didn't see the initialize func being called - Try:

```
const { initialize, isInitialized, discoverReaders } = useStripeTerminal();

// Call initialize on mount
useEffect(() => {
  void initialize();
}, []);

// Wait for initialization to complete before calling discoverReaders
useEffect(() => {
  if(isInitialized) {
    void discoverReaders();
  }
}, [isInitialized]);

Remember to implement error handling etc
```

**Comment by MoinJanjua** [comment]
I tried by this way too and have the same issue 
here is the error "StripeTerminalProvider component is not found, has not been mounted or SDK has not been initialized properly"

I was already initialized but facing the same issue

**Comment by mattlaroche-stripe** [comment]
I'm going to close out this issue as it has been about 10 months since the last update. If you have any other issues with this, please visit https://support.stripe.com/contact/email?topic=api_integration, click "I am a developer", and submit your issue, or contact us on [Discord](https://discord.com/invite/stripe).
